openupgradelib/openupgrade.py

```diff
@@ -3492,3 +3493,31 @@ def cow_templates_replicate_upstream(cr, mark_colname=None):
         """
         ).format(mark_identifier),
     )
+
+
+def clean_transient_models(cr):
+    """Clean transient models to prevent possible issues due to
+    chained data.
+
+    To be run at the base post-migration script for having a general scope.
+    Only works on > v8.
+
+    :param cr: Database cursor.
+    """
+    if version_info[0] < 9:
+        raise Exception("Not supported Odoo version for this method.")
+    cr.execute("SELECT model FROM ir_model WHERE transient")
+    table_names = [x[0] for x in cr.fetchall()]
+    for table_name in table_names:
+        try:
+            with cr.savepoint():
+                table = sql.Identifier(table_name)
+                query = sql.SQL(
+                    """DELETE FROM {} WHERE
+                    COALESCE(write_date, create_date, (now() at time zone 'UTC'))::timestamp
+                    < ((now() at time zone 'UTC') - interval '1 seconds')"""
+                ).format(table)
+                cr.execute(query)
+            cr.commit()  # pylint: disable=E8102
```

Comment:
Not sure If we should avoid this line.

Reply:
Yes, that would break incremental upgrade attempts during development/debugging of migration scripts. And is not necessary due to the saveoint.